packaging/gentoo/charliecloud-9999.ebuild

```diff
@@ -0,0 +1,75 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+# User namespaces don't play well with the sandbox.
+RESTRICT="test"
+
+if [[ ${PV} == "9999" ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/hpc/${PN}.git"
+	S="${WORKDIR}/${P}"
+else
+	SRC_URI="https://github.com/hpc/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="amd64 x86 ~amd64-linux ~x86-linux"
+	S="${WORKDIR}/${P}"
+fi
```

Comment:
That kind of conditional might break some package tooling in Gentoo. Typically the preferred solution here is to split the ebuild into two and thus avoiding the conditional blocks.

Reply:
That's extremely common practice, as I have learnt from Gentoo devs when proxy-maintaining packages. 
Also, I checked just now, and it's done for 5777 / 38657 of total ebuilds in Gentoo portage, and 1264 / 1954 of the live ebuilds. I would deduce from that that common practice is to not have separate ebuilds. 
So whatever tooling breaks, it must break for >65% of the official in-tree packages - could you tell me which tooling is broken by that? Up to now, I have not encountered any such issue, but was even adviced by Gentoo devs when pushing my proxy-maintained package in to have a common ebuild instead of two slightly different ones for live and release.

Reply:
Sure, if @monsieurp approved https://github.com/gentoo/gentoo/pull/6349, I am good :smile:

Reply:
Hang on a sec... I vouched for the opposite, namely: one live ebuild and one "normal" one. See https://github.com/gentoo/gentoo/pull/6349#issuecomment-348210928. Eventually, that's what @olifre did and I merged his contribution. re your first comment @nicolasbock: never heard of breakage stories due to an if conditional right in the middle of an ebuild to test the package version. It is indeed very common practice. Do you have a link to back up your claim? I'm curious now. :)

Reply:
My criticism was only related to the conditional block in the beginning of the ebuild and it goes back to a discussion on `gentoo-devel` where I was told that having such conditional "[...] is going to confuse the hell out of ekeyword" (https://archives.gentoo.org/gentoo-dev/message/015bb27eccec1da192bf550f18923693). And indeed, when I run `ekeyword` on this ebuild it changes both `KEYWORDS` assignments (in the `9999` branch _and_ in the `else` branch) which is maybe what @mgorny meant.

Reply:
@nicolasbock Thanks for pointing this out!
This issue with `ekeyword` is tracked here:
https://bugs.gentoo.org/614746
The workaround documented there already works for the updated Charliecloud-package in-tree (since I only have one `KEYWORDS` line in there). However, ideally, either `ekeyword` would see a fix or a check to `repoman` would be added and all ebuilds currently in-tree violating that would be updated...

Reply:
Nice workaround :smile:. Thanks for pointing out that bug. Hopefully this issue will get fixed soon in portage and the workaround will become obsolete.